test/Test/Toml/Gen.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE PatternSynonyms     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+
+-- | This module contains all generators for @tomland@ testing.
+
+module Test.Toml.Gen
+       ( genVal
+       , genKey
+       , genPrefixMap
+       , genToml
+       ) where
+
+import Control.Monad (forM)
+
+import Hedgehog (MonadGen)
+
+import Toml.PrefixTree (pattern (:||), Key (..), Piece (..), PrefixMap, PrefixTree (..), fromList)
+import Toml.Type (AnyValue (..), TOML (..), Value (..))
+
+import qualified Data.HashMap.Strict as HashMap
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+----------------------------------------------------------------------------
+-- Common generators
+----------------------------------------------------------------------------
+
+type V = Int
+
+genVal :: MonadGen m => m V
+genVal = Gen.int (Range.constant 0 256)
+
+-- TODO: Arrays and Date.
+-- | Generates random value of 'AnyValue' type.
+genAnyValue :: MonadGen m => m AnyValue
+genAnyValue = do
+  randB <- Gen.bool
+  randI <- toInteger <$> Gen.int (Range.constantBounded @Int)
+  randF <- Gen.double $ Range.constant @Double (-1000000.0) 1000000.0
+  randS <- Gen.text (Range.constant 0 256) Gen.alphaNum
+  Gen.element
+      [ AnyValue (Bool   randB)
+      , AnyValue (Int    randI)
+      , AnyValue (Float  randF)
+      , AnyValue (String randS)
+      ]
+
+-- TODO: unicode support
+genPiece :: MonadGen m => m Piece
+genPiece = Piece <$> Gen.text (Range.constant 1 50) Gen.alphaNum
+
+genKey :: MonadGen m => m Key
+genKey = Key <$> Gen.nonEmpty (Range.constant 1 10) genPiece
+
+genKeyAnyValue :: MonadGen m => m (Key, AnyValue)
+genKeyAnyValue = do
+    key <- genKey
+    val <- genAnyValue
+    pure (key, val)
```

Comment:
I think for such simple generators it's better to use shorter implementations with `liftA2` or similar.